package com.transsion.transfer.androidasync;

import java.nio.ByteBuffer;
import java.nio.channels.ReadableByteChannel;
import java.nio.channels.ScatteringByteChannel;
import java.nio.channels.SelectionKey;
import java.nio.channels.Selector;
import java.nio.channels.spi.AbstractSelectableChannel;

/* loaded from: classes6.dex */
abstract class o implements ReadableByteChannel, ScatteringByteChannel {

    /* renamed from: a, reason: collision with root package name */
    private AbstractSelectableChannel f55665a;

    /* JADX INFO: Access modifiers changed from: package-private */
    public o(AbstractSelectableChannel abstractSelectableChannel) {
        abstractSelectableChannel.configureBlocking(false);
        this.f55665a = abstractSelectableChannel;
    }

    @Override // java.nio.channels.Channel, java.io.Closeable, java.lang.AutoCloseable
    public void close() {
        this.f55665a.close();
    }

    public boolean d() {
        return false;
    }

    public abstract boolean h();

    public abstract SelectionKey i(Selector selector);

    @Override // java.nio.channels.Channel
    public boolean isOpen() {
        return this.f55665a.isOpen();
    }

    public SelectionKey k(Selector selector, int i11) {
        return this.f55665a.register(selector, i11);
    }

    public abstract void l();

    public abstract int m(ByteBuffer[] byteBufferArr);
}
